tests/kernel_performance/hanaperf_postinstall.pm

```diff
@@ -0,0 +1,153 @@
+        $ver_cfg = "PRODUCT_RELEASE=SLES-" . get_var('VERSION') . ";PRODUCT_BUILD=$mybuild";
+        record_info($ver_cfg);
+    }
+
+    # Disable service
+    assert_script_run("systemctl disable $disable_services");
+
+    # sync time
+    assert_script_run("chronyd -q 'server 0.europe.pool.ntp.org iburst'");
+
+    # set static hostname
+    assert_script_run("hostnamectl hostname `hostname -s`");
+
+    # create basic /root/qaset/config
+    assert_script_run("mkdir -p /root/qaset/");
+    my $qaset_config_file = <<'EOF';
+_QASET_ROLE=HANA
+SQ_TEST_RUN_SET=performance
+SQ_MSG_QUEUE_ENALBE=y
+_QASET_SOFTWARE_TAG=baremetal
+_QASET_SOFTWARE_SUB_TAG=default
+EOF
+    assert_script_run("echo -n '$qaset_config_file' > /root/qaset/config");
+    assert_script_run("echo '_QASET_KERNEL_TAG=$qaset_kernel_tag' >> /root/qaset/config") if $qaset_kernel_tag ne '';
+
+    # The qaset/config need not be updated when role is HANA and ABAP
+    return if (get_var('PROJECT_M_ROLE', "") =~ /PROJECT_M_HANA|PROJECT_M_ABAP/);
+
+    # Extract the openQA parameter: QASET_CONFIG and VER_CFG
+    # Historical reasons that the different settings splitted into two parameters.
+    # VER_CFG="PRODUCT_RELEASE=SLES-15-SP3;PRODUCT_BUILD=202109"
+    extract_settings_qaset_config(get_var("QASET_CONFIG", ''));
+    extract_settings_qaset_config($ver_cfg);
+}
+
+sub os_update {
+    my $update_repo_url = shift;
+    my $zypper_repo_path = "/etc/zypp/repos.d";
+
+    assert_script_run("wget -N -P $zypper_repo_path $update_repo_url 2>&1");
+    zypper_call("--gpg-auto-import-keys ref");
+    zypper_call("dup", timeout => 1800);
+}
+
+sub handle_repo_and_package {
+    my $package_install = "qa_lib_ctcs2 wget bc bzip2 screen cpupower pciutils lsscsi smartmontools netcat-openbsd libltdl7 unzip lvm hana_insserv_compat";
+    my $package_remove = "snapper-zypp-plugin";
+
+    # Add QA:HEAD repo
+    add_qa_head_repo;
+
+    # Install mandatory packages for SLE16
+    zypper_call("install $package_install");
+    zypper_call("rm $package_remove");
+}
+
+sub run {
+    my $self = shift;
+
+    my $project_m_role = get_var("PROJECT_M_ROLE", "");
+
+    select_console 'root-console';
+
+    # Add repo, remove and install packages
+    handle_repo_and_package;
+
+    # Update OS for MU testing
+    if (my $hana_perf_os_update = get_var("HANA_PERF_OS_UPDATE")) {
+        os_update($hana_perf_os_update);
+    }
+
+    # Install SLEperf test framework
+    install_sleperf;
+
+    # Setup environment
+    setup_environment;
+
+    # Hold and wait other job finish if role is driver
+    wait_for_children if ($project_m_role eq "PROJECT_M_DRIVER");
+
+    # Reboot system
+    power_action('reboot', textmode => 1);
+    if (is_x86_64) {
+        # Handle x86_64 bare-metal reboot
+        switch_from_ssh_to_sol_console(reset_console_flag => 'on');
+        sleep 30;
+        assert_screen("linux-login", 2400);
```

Comment:
Same suggestion about double and single quotes.

Reply:
thanks! fixed.